lucet-runtime/lucet-runtime-internals/src/c_api.rs

```diff
@@ -182,13 +185,15 @@ impl From<lucet_alloc_limits> for Limits {
 
 impl From<&lucet_alloc_limits> for Limits {
     fn from(limits: &lucet_alloc_limits) -> Limits {
-        Limits {
-            heap_memory_size: limits.heap_memory_size as usize,
-            heap_address_space_size: limits.heap_address_space_size as usize,
-            stack_size: limits.stack_size as usize,
-            globals_size: limits.globals_size as usize,
-            signal_stack_size: limits.signal_stack_size as usize,
-        }
+        Limits::default()
+            .with_heap_memory_size(limits.heap_memory_size as usize)
+            .with_heap_address_space_size(limits.heap_address_space_size as usize)
+            .with_stack_size(limits.stack_size as usize)
+            .expect("lucet_alloc_limts has a valid stack size")
```

Comment:
The `Limits` changes also make the C API a little more panicky, but we can't enforce the same consistency in building a `lucet_alloc_limits`, so this is the first place to find an invalid setting.

Reply:
This should be resolved by moving the check to `validate`